src/databricks/labs/lsql/dashboards.py

```diff
@@ -304,82 +294,31 @@ def from_path(cls, path: Path) -> "TileMetadata":
         return cls.from_dict(header)
 
     def __repr__(self):
-        return f"TileMetadata<{self._path}>"
+        return f"TileMetadata<{self.id}>"
 
 
 @dataclass
-class DashboardMetadata:
-    display_name: str
-    tiles: dict[str, TileMetadata] = dataclasses.field(default_factory=dict)
-
-    @classmethod
-    def from_dict(cls, raw: dict) -> "DashboardMetadata":
-        display_name = raw["display_name"]  # Fail early if missing
-        tiles, tiles_raw = {}, raw.get("tiles", {})
-        for tile_id, tile_raw in tiles_raw.items():
-            if not isinstance(tile_raw, dict):
-                logger.warning(f"Parsing invalid tile metadata in dashboard.yml: tiles.{tile_id}.{tile_raw}")
-                continue
-            tile = TileMetadata(id=tile_id)
-            for tile_key, tile_value in tile_raw.items():
-                if tile_key == "id":
-                    logger.warning(f"Parsing unsupported field in dashboard.yml: tiles.{tile_id}.id")
-                    continue
-                try:
-                    tile_new = TileMetadata.from_dict({tile_key: tile_value})
-                    tile.update(tile_new)
-                except TypeError:
-                    logger.warning(f"Parsing unsupported field in dashboard.yml: tiles.{tile_id}.{tile_key}")
-                    continue
-            tiles[tile.id] = tile
-        return cls(display_name=display_name, tiles=tiles)
-
-    def as_dict(self) -> dict:
-        raw: dict = {"display_name": self.display_name}
-        if self.tiles:
-            raw["tiles"] = {tile.id: tile.as_dict() for tile in self.tiles.values()}
-        return raw
-
-    @classmethod
-    def from_path(cls, path: Path) -> "DashboardMetadata":
-        """Export dashboard metadata from a YAML file."""
-        fallback_metadata = cls(display_name=path.parent.name)
-        if not path.exists():
-            return fallback_metadata
-        try:
-            raw = yaml.safe_load(path.read_text())
-        except yaml.YAMLError as e:
-            logger.warning(f"Parsing {path}: {e}")
-            return fallback_metadata
-        try:
-            return cls.from_dict(raw)
-        except KeyError as e:
-            logger.warning(f"Parsing {path}: {e}")
-            return fallback_metadata
-
-
 class Tile:
     """A dashboard tile."""
 
-    def __init__(self, tile_metadata: TileMetadata) -> None:
-        self._metadata = tile_metadata
+    metadata: TileMetadata
 
-        default_width, default_height = self._default_size()
-        width = self._metadata.width or default_width
-        height = self._metadata.height or default_height
-        self.position = Position(0, 0, width, height)
+    _position: Position = Position(0, 0, 0, 0)
```

Comment:
```suggestion
    position: Position = Position(0, 0, 0, 0)
```
otherwise looks a bit ugly... rename `def position` to `def position_or_default`